### PR TITLE
BridgeCrew compliance updates

### PR DESCRIPTION
**Pull request by mclacore**
This is an identical PR to https://github.com/massdriver-cloud/azure-cosmosdb-mongo/pull/20

closes: https://github.com/massdriver-cloud/massdriver/issues/1374

### CODE REVIEW

File: src/_providers.tf [hunk 1]
```diff
@@ -24,3 +28,9 @@ provider "azurerm" {
   client_secret   = var.azure_service_principal.data.client_secret
   subscription_id = var.azure_service_principal.data.subscription_id
 }
+
+provider "azuread" {
```
Comment (coryodaniel) [inline src/_providers.tf:32]: @chrisghill will need to confirm, i'm not positive if we have this provider loaded in XO.

File: src/cmk.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      "Import",
+      "WrapKey",
+      "UnwrapKey"
+    ]
+  }
+
+  tags = var.md_metadata.default_tags
+}
+
+resource "azurerm_key_vault_key" "main" {
```
[the remark this answers is not in the document]
Comment (mclacore) [reply src/cmk.tf:57]: @chrisghill how should we approach this critical compliance? At some point a customer's key will expire with this enabled. Do we want to set up auto-rotate or some type of notification?

[the remark this answers is not in the document]
Comment (mclacore) [reply src/cmk.tf:58]: Looks like it's not currently supported in AzureRM Terraform, but should be soon: https://github.com/hashicorp/terraform-provider-azurerm/issues/14471

File: src/_providers.tf (hunk 1 above)
Comment (jaketf) [reply src/_providers.tf:32]: @coryodaniel I feel like it is we used it in the databricks bundle which worked fine for mckinsey pilot

Comment (jaketf) [reply src/_providers.tf:32]: and it appears in cached providers here: https://github.com/massdriver-cloud/massdriver-bundles/blob/main/provisioners/terraform/cached-providers/providers.tf#L28-L30

is there another reason i'm not aware of that it wouldn't be in XO?

**Review by coryodaniel — APPROVED** [review]
LGTm.

**Pull request merged**